### PR TITLE
PL 129843 Nginx should be fully configurable via Nix code

### PR DESCRIPTION
**Pull request by mkg20001**
**upstream sync moved to PR #349**

@flyingcircusio/release-managers

## Release process

Impact:

Changelog:
- flyingcircus nginx configurations can now be written in pure nix language now

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - no security related changes
- [x] Security requirements tested? (EVIDENCE)

### CODE REVIEW

File: nixos/services/nginx/default.nix [hunk 1]
```diff
@@ -101,7 +102,7 @@ let
 
   baseHttpConfig = ''
     # === Defaults ===
-    default_type application/octet-stream;
+    # default_type application/octet-stream;
```
Comment (dpausp) [inline nixos/services/nginx/default.nix]: Do we need this?

**Review by dpausp — CHANGES_REQUESTED** [review]
In the JSON config, setting an *SSL option automatically enables enableACME. We want to keep that. Doesn't work with the new NixOS options so the example config doesn't build anymore.

File: nixos/services/nginx/default.nix (hunk 1 above)
Comment (mkg20001) [reply nixos/services/nginx/default.nix]: Not really, removing

**Comment by mkg20001** [comment]
Indeed strange, as https://github.com/flyingcircusio/fc-nixos/pull/340/files#diff-f5fb9002fd859a9923715a59f5f83653e6baf9675b48013b8dc5bd567661ec75L73 should've enabled it.

**Comment by mkg20001** [comment]
image

Trying the example config from the README.txt results in acme jobs for the domain - couldn't reproduce

**Comment by mkg20001** [comment]
Tried with the nix version, that one indeed causes the problem. The issue is that `//` doesn't properly work with things from the module system, using lib.mkMerge causes other trouble (we have to override using mkForce, which means that if the user specifies enableACME themselves to false it will be overwritten, too)

**Comment by mkg20001** [comment]
Fixed. Now using the module system properly and overriding defaults for both enableACME and listenAddresses with what the fc function previously did.

**Pull request merged**